tests/checker/good/gold/dead_code_test13.scilla.gold

```diff
@@ -0,0 +1,139 @@
+{
+  "cashflow_tags": { "State variables": [], "ADT constructors": [] },
+  "contract_info": {
+    "scilla_major_version": "0",
+    "vname": "Dead13",
+    "params": [],
+    "fields": [],
+    "transitions": [ { "vname": "foo", "params": [] } ],
+    "procedures": [],
+    "events": [],
+    "ADTs": [
+      {
+        "tname": "dead_code_test13.C",
+        "tparams": [],
+        "tmap": [ { "cname": "dead_code_test13.C", "argtypes": [] } ]
+      },
+      {
+        "tname": "dead_code_test13.B",
+        "tparams": [],
+        "tmap": [ { "cname": "dead_code_test13.B1", "argtypes": [] } ]
+      },
+      {
+        "tname": "Option",
+        "tparams": [ "'A" ],
+        "tmap": [
+          { "cname": "Some", "argtypes": [ "'A" ] },
+          { "cname": "None", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Bool",
+        "tparams": [],
+        "tmap": [
+          { "cname": "True", "argtypes": [] },
+          { "cname": "False", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Nat",
+        "tparams": [],
+        "tmap": [
+          { "cname": "Zero", "argtypes": [] },
+          { "cname": "Succ", "argtypes": [ "Nat" ] }
+        ]
+      },
+      {
+        "tname": "List",
+        "tparams": [ "'A" ],
+        "tmap": [
+          { "cname": "Cons", "argtypes": [ "'A", "List ('A)" ] },
+          { "cname": "Nil", "argtypes": [] }
+        ]
+      },
+      {
+        "tname": "Pair",
+        "tparams": [ "'A", "'B" ],
+        "tmap": [ { "cname": "Pair", "argtypes": [ "'A", "'B" ] } ]
+      },
+      {
+        "tname": "dead_code_test13.A",
+        "tparams": [],
+        "tmap": [
+          { "cname": "dead_code_test13.A1", "argtypes": [] },
+          { "cname": "dead_code_test13.A2", "argtypes": [] }
+        ]
+      }
+    ]
+  },
+  "warnings": [
+    {
+      "warning_message": "Unused ADT constructor: A2",
+      "start_location": {
+        "file": "contracts/dead_code_test13.scilla",
+        "line": 7,
+        "column": 1
+      },
+      "end_location": { "file": "", "line": 0, "column": 0 },
+      "warning_id": 3
+    },
+    {
+      "warning_message": "Unused ADT constructor: B1",
+      "start_location": {
+        "file": "contracts/dead_code_test13.scilla",
+        "line": 11,
+        "column": 1
+      },
+      "end_location": { "file": "", "line": 0, "column": 0 },
+      "warning_id": 3
+    },
+    {
+      "warning_message": "Unused ADT constructor: C",
```

Comment:
This actually tests for false negative described in #1125.

Unfortunately, there are no such cases on the reference smart contracts set. So, any ideas how to break this logic on more complicated cases are more than welcome.